File: ui_ux/lib/models/student.dart
class Student {
  final String fullName;
  final int classStudies;
  final String location;
  final String phoneNumber;
  final String password;
  final String picturePath;
  final String institution;
  final List<String> teacherId;

  Student({
    required this.fullName,
    required this.classStudies,
    required this.location,
    required this.phoneNumber,
    required this.password,
    required this.picturePath,
    required this.institution,
    required this.teacherId,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      fullName: json['fullName'],
      classStudies: json['classStudies'],
      location: json['location'],
      phoneNumber: json['phoneNumber'],
      password: json['password'],
      picturePath: json['picturePath'],
      institution: json['institution'],
      teacherId: List<String>.from(json['teacherId']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'classStudies': classStudies,
      'location': location,
      'phoneNumber': phoneNumber,
      'password': password,
      'picturePath': picturePath,
      'institution': institution,
      'teacherId': teacherId,
    };
  }
}

File: ui_ux/backend/models/Student.js
const mongoose = require("mongoose");
const bcrypt = require("bcrypt");

const Schema = mongoose.Schema;

const studentSchema = new Schema({
  fullName: {
    type: String,
    required: true,
    min: 2,
    max: 50,
  },

  classStudies: {
    type: String, //will be string
    required: true,
  },
  location: {
    type: String,
    required: true,
    max: 50,
  },
  phoneNumber: {
    type: String,
    required: true,
    min: 6,
    max: 12,
  },
  // password: {
  //   type: String,
  //   required: true,
  //   min: 5,
  // },
  picturePath: {
    type: String,
    default: "",
    required: false,
  },
  institution: {
    type: String,
    max: 30,
    required: false,
  },
  teacherId: {
    type: Array,
    //required: true,
    default: [],
  },
  advertisementId: {
    type: Array,
    default: [],
  },
  email: {
    type: String,
    required: true,
    unique: true,
  },
});

module.exports = mongoose.model("Student", studentSchema);
